Fix duplicated error notices

malithsen opened the pull request:
`StoreNoticeProvider` is already being used within `Block` component. This PR proposes its removal from `FrontendBlock` to prevent duplicate error notices mentioned in  #5475 . I am not quite sure if it should rather be removed from the (child) `Block` component instead.  

<!-- Start by describing the changes made in this Pull Request, and the reason for such changes. -->

<!-- Reference any related issues or PRs here -->
Fixes #5475 


<!-- Don't forget to update the title with something descriptive. -->
<!-- If your pull request implements a feature flag, make sure you update [this doc](../docs/blocks/features-and-blocks-behind-a-flag.md) -->

### Screenshots

**Before**
![image](https://user-images.githubusercontent.com/41606954/141514500-4803d148-6a73-4ea5-aefd-a24f9a6cbc7b.png)

**After**
![Screen Shot 2021-12-30 at 12 15 43 PM](https://user-images.githubusercontent.com/6216000/147777983-9f54e997-0fbb-40fa-a01c-ac3648d34965.png)


### Testing

How to test the changes in this Pull Request:


1. Make sure you're using the Blocks checkout and WCPay.
2. Enable UPE from WCPay Dev Tools
2. Add 1 or more products to your cart.
3. Go to the blocks checkout page
4. Pay through the credit card form with a card that will be declined, e.g. 4000 0000 0000 0002.
5. Ensure that only one error notice is displayed 

### User Facing Testing
These are steps for user testing (where "user" is someone interacting with this change that is not editing any code).
* [x] Same as above

### Changelog

> Fix duplicated checkout error notices.

malithsen committed:
Fix duplicated error notices

StoreNoticeProvider is already being used within Block component. This
removal prevents duplicated error notices within the Payment Options
section

opr reviewed (changes requested):
I think there's a _slight_ performance benefit for removing it from the child block as you pointed out and leaving it in the frontend block.

Doing it this way `StoreNoticesProvider` won't load in the editor. It's not needed there as we don't show notices in the editor.

I also had to rebase this on trunk to fix an issue with the country code being passed through to WC pay which was fixed earlier.

malithsen committed:
Move StoreNoticeProvider from payment block

Remove StoreNoticeProvider from payment block and move it to the parent
frontend block to avoid unnecessarily loading the notice provider

malithsen commented:
Thanks for pointing out the performance improvement @opr

- Rebased on trunk
- Moved `StoreNoticeProvider` to frontend-block from block https://github.com/woocommerce/woocommerce-gutenberg-products-block/pull/5476/commits/149461f985832b964e084eaf5755a287b651a3c5

opr reviewed (approved):
Great, thank you for making these changes this seems to be working well!
